### ISSUE TITLE
Reading of model budget files goes wrong for DISV model with inactive cells

### ISSUE THREAD

**Issue by HendrikKok** [Deltares/imod-python#1247]
The open_cbc(flowja=False) crashes on the model budgets of DISV model. Something goes wrong in the disv_to_horizontal_lower_indices(). There seems to be a shape mismatch. The method works is flowja=True. It has maybe something to do with inactive cells?

Can easily be tested on the test_simulation_write_disv test in the UZFf tests. Just remove the flowja=True

**Comment by Huite** [comment]
What do you mean crash? I'm guessing you get an error report, right? Probably an IndexError?

**Comment by Huite** [comment]
I've had a look, but I don't immediately see an issue yet, even with the clue of having an inactive cell.

My assumption is that ia and ja use the external numbering. In that case, inactive cells should simply have zero connections in the CSR adjacency matrix. ncells should also be inclusive of inactive cells, so it's not clear to me.

While I am looking it, I wonder if it needs numba though?

Wouldn't this work as well?

```python
def disv_lower_index(
    ia: IntArray,
    ja: IntArray,
    ncells: int,
    nlayer: int,
    ncells_per_layer: int,
) -> IntArray:
    lower = np.full(ncells, -1, np.int64)
    i = expand_indptr(ia)
    j = ja - 1
    d = j - i
    is_vertical = d >= ncells_per_layer
    index = np.arange(j.size)
    i_lower = i[is_vertical + (d[is_vertical] - ncells_per_layer)]
    lower[i_lower] = index[is_vertical]
    return lower.reshape(nlayer, ncells_per_layer)
```

(Checking of the logic is left as an exercise to the reader)

With a vectorized function, we can merge it with `disv_horizontal_index`.

**Comment by HendrikKok** [comment]
I don't think that works. To start with the fact that i and j are not of the same shape right? The follow example seems to work for me.

'''
def disv_lower_index(
    ia: IntArray,
    ja: IntArray,
    ncells: int,
    nlayer: int,
    ncells_per_layer: int,
) -> IntArray:
    lower = np.full(ncells, -1, np.int64)
    ncon = np.diff(ia)
    i = np.repeat(np.arange(ncon.size), ncon)
    j = np.arange(ja.size)
    d = ja - (i + 1)
    is_lower_vertical = d >= ncells_per_layer
    lower[i[is_lower_vertical]] = j[is_lower_vertical]
    return lower.reshape(nlayer, ncells_per_layer)
'''

**Comment by Huite** [comment]
`ia` and `ja` are not of the same size, but `i` and `j` are (we're going from CSR to COO sparse matrix form, should be added in a comment I!)

And we still need the offset for the IDOMAIN -1 cells. But I'll have a quick look at it!

**Comment by Huite** [comment]
I'm guessing I found at least one mismatch in `disv_horizontal_index`:

```python
    is_connection = edge_face_connectivity[:, 1] != fill_value
    edge_faces = edge_face_connectivity[is_connection]
    order = np.argsort(np.lexsort(edge_faces.T[::-1]))
    # Reshuffle for every layer
    index = index[:, order]
```

edge_faces will contain all edges. ia, ja or (i, j) will only represent edges where i and j are both active.

**Comment by JoerivanEngelen** [comment]
Fixed in #1250 (FYI: issue was not auto-closed,  because PR was not merged directly to master)
